docs/src/index.md

```diff
@@ -449,27 +451,36 @@ x2   -0.152062  0.124931 -1.21717   0.2582
 
 This model can also be fit as
 ```jldoctest
-julia> lm(hcat(ones(10), 1:10), randn(MersenneTwister(12321), 10))
-GLM.LinearModel{GLM.LmResp{Array{Float64,1}},GLM.DensePredChol{Float64,Base.LinAlg.Cholesky{Float64,Array{Float64,2}}}}:
+julia> using Random; Random.seed!(12321);
+
+julia> lm(hcat(ones(10), 1:10), randn(10))
+LinearModel{LmResp{Array{Float64,1}},DensePredChol{Float64,LinearAlgebra.Cholesky{Float64,Array{Float64,2}}}}:
 
 Coefficients:
       Estimate Std.Error  t value Pr(>|t|)
 x1    0.717436  0.775175 0.925515   0.3818
 x2   -0.152062  0.124931 -1.21717   0.2582
 ```
 
-### Methods for model updating
+### Model methods
 ```@docs
+GLM.cancancel
 delbeta!
+StatsBase.deviance
+GLM.dispersion
+GLM.installbeta!
+GLM.issubmodel
 linpred!
 linpred
-GLM.installbeta!
-GLM.cancancel
+lm
+StatsBase.nobs
+StatsBase.nulldeviance
+StatsBase.predict
 updateμ!
 wrkresp
 GLM.wrkresp!
-GLM.dispersion
 ```
+<!-- ftest should be included as well but currently Documenter can't parse the doctest for ftest-->
```

Comment:
Any idea why?

Reply:
No. I tried to figure out what was going on but I just think it is some kind of parser bug in Documenter. I'm getting
```julia
[ Info: CheckDocument: running document checks.
┌ Warning: failed to parse exception in `src/index.md`
│   exception = Base.Meta.ParseError("extra token \"DOF\" after end of expression")
└ @ Documenter.Utilities ~/.julia/packages/Documenter/zOHWz/src/Utilities/Utilities.jl:89
=====[Test Error]==============================

> Location: /Users/andreasnoack/.julia/dev/GLM/src/ftest.jl:57-79

> Code block:

```jldoctest
julia> dat = DataFrame(Treatment=[1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2.],
                              Result=[1.1, 1.2, 1, 2.2, 1.9, 2, .9, 1, 1, 2.2, 2, 2],
                              Other=[1, 1, 2, 1, 2, 1, 3, 1, 1, 2, 2, 1]);

julia> model = lm(@formula(Result ~ 1 + Treatment), dat);

julia> nullmodel = lm(@formula(Result ~ 1), dat);

julia> bigmodel = lm(@formula(Result ~ 1 + Treatment + Other), dat);

julia> ft = ftest(model.model, nullmodel.model)
        Res. DOF DOF ΔDOF    SSR    ΔSSR     R²    ΔR²       F* p(>F)
Model 1       10   3      0.1283         0.9603
Model 2       11   2   -1 3.2292 -3.1008 0.0000 0.9603 241.6234 <1e-7


julia> ftest(bigmodel.model, model.model, nullmodel.model)
        Res. DOF DOF ΔDOF    SSR    ΔSSR     R²    ΔR²       F*  p(>F)
Model 1        9   4      0.1038         0.9678
Model 2       10   3   -1 0.1283 -0.0245 0.9603 0.0076   2.1236 0.1790
Model 3       11   2   -1 3.2292 -3.1008 0.0000 0.9603 241.6234  <1e-7
```

> Subexpression:

    ft = ftest(model.model, nullmodel.model)
     Res. DOF DOF ΔDOF    SSR    ΔSSR     R²    ΔR²       F* p(>F)

> Output Diff:

        Res. DOF DOF ΔDOF    SSR    ΔSSR     R²    ΔR²       F* p(>F)
Model 1       10   3      0.1283         0.9603
Model 2       11   2   -1 3.2292 -3.1008 0.0000 0.9603 241.6234 <1e-7

=====[End Error]===============================
┌ Warning: failed to parse exception in `src/index.md`
│   exception = Base.Meta.ParseError("extra token \"DOF\" after end of expression")
└ @ Documenter.Utilities ~/.julia/packages/Documenter/zOHWz/src/Utilities/Utilities.jl:89
=====[Test Error]==============================

> Location: /Users/andreasnoack/.julia/dev/GLM/src/ftest.jl:57-79

> Code block:

```jldoctest
julia> dat = DataFrame(Treatment=[1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2.],
                              Result=[1.1, 1.2, 1, 2.2, 1.9, 2, .9, 1, 1, 2.2, 2, 2],
                              Other=[1, 1, 2, 1, 2, 1, 3, 1, 1, 2, 2, 1]);

julia> model = lm(@formula(Result ~ 1 + Treatment), dat);

julia> nullmodel = lm(@formula(Result ~ 1), dat);

julia> bigmodel = lm(@formula(Result ~ 1 + Treatment + Other), dat);

julia> ft = ftest(model.model, nullmodel.model)
        Res. DOF DOF ΔDOF    SSR    ΔSSR     R²    ΔR²       F* p(>F)
Model 1       10   3      0.1283         0.9603
Model 2       11   2   -1 3.2292 -3.1008 0.0000 0.9603 241.6234 <1e-7


julia> ftest(bigmodel.model, model.model, nullmodel.model)
        Res. DOF DOF ΔDOF    SSR    ΔSSR     R²    ΔR²       F*  p(>F)
Model 1        9   4      0.1038         0.9678
Model 2       10   3   -1 0.1283 -0.0245 0.9603 0.0076   2.1236 0.1790
Model 3       11   2   -1 3.2292 -3.1008 0.0000 0.9603 241.6234  <1e-7
```

> Subexpression:

    ftest(bigmodel.model, model.model, nullmodel.model)
     Res. DOF DOF ΔDOF    SSR    ΔSSR     R²    ΔR²       F*  p(>F)

> Output Diff:

        Res. DOF DOF ΔDOF    SSR    ΔSSR     R²    ΔR²       F*  p(>F)
Model 1        9   4      0.1038         0.9678
Model 2       10   3   -1 0.1283 -0.0245 0.9603 0.0076   2.1236 0.1790
Model 3       11   2   -1 3.2292 -3.1008 0.0000 0.9603 241.6234  <1e-7
```

Reply:
cc @mortenpi

Reply:
If I remember correctly, we have a simple heuristic in Documenter such that if the next line after the first `julia>` line has 6 or more spaces, it is treated as a continuation to the source code, and not as output. It is technically valid input to the REPL:

```
julia> ft = ftest(model.model, nullmodel.model)
         Res. DOF DOF ΔDOF    SSR    ΔSSR     R²    ΔR²       F* p(>F)
ERROR: syntax: extra token "DOF" after end of expression
```

I don't have an idea right away how to solve it though, but an issue would be welcome.

Reply:
Ah, that's annoying. What do you mean by "its a valid input to the REPL"? When I paste this block in the REPL, only the first line is kept (without the `julia> ` prefix).

Maybe Documenter could count opening and closing parentheses, and consider that a line is finished if all opening parens have been closed? That should work for most cases, and when it doesn't detect multiline commands people could still add parentheses. Or would it be possible to use the same mechanism as the REPL?